qiskit_nature/problems/sampling/protein_folding/protein_folding_problem.py

```diff
@@ -10,17 +10,22 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 """Defines a protein folding problem that can be passed to algorithms."""
-from typing import Union, List
+from __future__ import annotations
 
-from qiskit.opflow import PauliSumOp, PauliOp
+from typing import Union, List, TYPE_CHECKING
 
+from qiskit.opflow import PauliSumOp, PauliOp
+from qiskit.algorithms import MinimumEigensolverResult
 from .peptide.peptide import Peptide
 from .interactions.interaction import Interaction
 from .penalty_parameters import PenaltyParameters
 from .qubit_op_builder import QubitOpBuilder
 from .qubit_utils import qubit_number_reducer
 from ..sampling_problem import SamplingProblem
 
+if TYPE_CHECKING:
+    from ....results.protein_folding_result import ProteinFoldingResult
```

Comment:
```suggestion
    from qiskit_nature.results.protein_folding_result import ProteinFoldingResult
```
We typically only use relative imports within the same sub-module